commit dcd505d9801990781123093de5913f365b94782f
Author: Halton Huo <halton.huo@sun.com>
Date:   Fri Sep 25 14:46:27 2009 +0800

    Correct tty return value after VT support is integrated in Solaris.

diff --git a/src/ck-sysdeps-solaris.c b/src/ck-sysdeps-solaris.c
index bc3b695..607a3ae 100644
--- a/src/ck-sysdeps-solaris.c
+++ b/src/ck-sysdeps-solaris.c
@@ -180,27 +180,8 @@ stat2proc (pid_t        pid,
 	if (tty_maj == 15) {
 		snprintf (P->tty_text, sizeof P->tty_text, "/dev/vt/%u", tty_min);
         }
-        if (tty_maj == 24) {
-                snprintf (P->tty_text, sizeof P->tty_text, "/dev/pts/%u", tty_min);
-        }
-
         if (P->tty == NO_TTY_VALUE) {
-#ifdef HAVE_SYS_VT_H
                 memcpy (P->tty_text, "   ?   ", 8);
-#else
-                /*
-                 * This is a bit of a hack.  On Solaris, pre-VT integration, the
-                 * Xorg process is not assigned a TTY.  So, just assign the value
-                 * to "/dev/console" if running without VT support.  This will
-                 * allow people using Solaris pre-VT integration to use
-                 * ConsoleKit.
-                 */
-                memcpy (P->tty_text, "/dev/console", 12);
-#endif
-        }
-
-        if (P->tty == DEV_ENCODE(0,0)) {
-                memcpy (P->tty_text, "/dev/console", 12);
         }
 
         if (P->pid != pid) {
